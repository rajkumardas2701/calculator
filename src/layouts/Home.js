import NavBar from './NavBar';

const Home = () => (
  <div className="AppBody">
    <NavBar />
    <h3 data-testid="home-test">Welcome to our page!</h3>
    <p data-testid="home-para-test">
      Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt
      ut labore et dolore magna aliqua. Utenim ad minim veniam, quis nostrud exercitation ullamco
      laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor inreprehenderit in
      voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat
      non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
    </p>
    <br />
    <p>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt
      ut labore et dolore magna aliqua. Utenim ad minim veniam, quis nostrud exercitation ullamco
      laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor inreprehenderit in
      voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat
      non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
    </p>
  </div>
);
export default Home;
