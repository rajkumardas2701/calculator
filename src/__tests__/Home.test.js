import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Home from '../layouts/Home';

afterEach(cleanup);

it('should display the heading', () => {
  const { getByTestId } = render(<Home />);
  expect(getByTestId('home-test')).toHaveTextContent('Welcome to our page!');
});

it('should display the first text paragraph', () => {
  const { getByTestId } = render(<Home />);
  expect(getByTestId('home-para-test')).toHaveTextContent(/Lorem ipsum/i);
});

it('renders correctly', () => {
  const home = renderer.create(<Home />).toJSON();
  expect(home).toMatchSnapshot();
});
