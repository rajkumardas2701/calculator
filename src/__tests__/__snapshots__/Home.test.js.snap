// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="AppBody"
>
  <nav>
    <h1
      data-testid="navBar-header"
    >
      Math Magicians
    </h1>
    <div
      className="rightNav"
      data-testid="navBar-test"
    >
      <div>
        <a
          href="/"
        >
          Home
        </a>
      </div>
      <div>
        <a
          href="/calculator"
        >
          Calculator
        </a>
      </div>
      <div>
        <a
          href="/quote"
        >
          Quotes
        </a>
      </div>
    </div>
  </nav>
  <h3
    data-testid="home-test"
  >
    Welcome to our page!
  </h3>
  <p
    data-testid="home-para-test"
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Utenim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor inreprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
  </p>
  <br />
  <p>
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Utenim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor inreprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
  </p>
</div>
`;
